tests/kernel/kselftests.pm

```diff
@@ -14,24 +14,39 @@ use testapi;
 use warnings;
 use serial_terminal 'select_serial_terminal';
 use LTP::WhiteList;
+use version_utils qw(is_transactional);
+use transactional 'trup_install';
+
+sub download_kernel_source
+{
+    my @kv = split /\./, script_output "uname -r";
+    my ($kv0, $kv1, $kv2) = ($kv[0], $kv[1], (split /-/, $kv[2])[0]);
+    my $url = "https://mirrors.edge.kernel.org/pub/linux/kernel/v$kv0.x/linux-$kv0.$kv1";
```

Comment:
IMHO https://mirrors.kernel.org/ is the site (https://mirrors.edge.kernel.org/ is a redirection which can change one day), it's recommended in https://mirrors.edge.kernel.org/.